twine/commands/upload.py

```diff
@@ -127,6 +127,14 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     repository = upload_settings.create_repository()
     uploaded_packages = []
 
+    # If a user tries to upload signature files with no actual distributions,
```

Comment:
I generally like adding comments above conditional logic to make it more apparent what it does. That said, I've removed it since the exception message can explain it.

Reply:
Comments that just tell you what the code is doing when it's not incredibly baroque aren't helpful as they're more likely to drift out-of-date with the actual code. A comment that might be helpful is "Ensure we don't silently fail if a user tries to upload _just_ an signature file" but even that's questionable since the condition is clear and the exception message explains exactly what the user did wrong.